docs: add setup anoncreds and indy-vdr (#107)
animo/aries-javascript-docs: guides/getting-started/set-up/anoncreds-rs.md

When using Aries Framework JavaScript with AnonCreds RS, there are a few extra dependencies that need to be installed. We need to install `@hyperledger/anoncreds` package, which contains the interfaces, and `@hyperledger/anoncreds-rs` package which is an implementation which depends on a wrapper of anoncreds-rs. Secondly, we need to add native bindings for the specific platform `@hyperledger/anoncreds-<platform>`. Currently there are bindings for Node.JS, as `@hyperledger/anoncreds-nodejs`, and React Native as `@hyperlegder/anoncreds-react-native`.

<!--tabs-->

# Node.JS

```console
yarn add @aries-framework/anoncreds @aries-framework/anoncreds-rs @hyperledger/anoncreds-nodejs
```

# React Native

```console
yarn add @aries-framework/anoncreds @aries-framework/anoncreds-rs @hyperledger/anoncreds-react-native
```

<!--/tabs-->

### Adding AnonCreds RS to the agent

After installing the dependencies, we should register both the `AnonCreds` and `AnonCredsRs` module on the agent.

<!--tabs-->

# Node.JS

```typescript showLineNumbers set-up-anoncreds-rs.ts section-1

```

# React Native

```typescript showLineNumbers set-up-anoncreds-rs-rn.ts section-1

```

<!--/tabs-->

> As you can see, the AnonCreds module takes a list of registry modules. These modules will be used to resolve the AnonCreds objects. We recommend [Indy VDR](./indy-vdr) as an AnonCreds registry for Hyperledger Indy networks, but [Indy SDK](./indy-sdk) also be used.

Reviewer: ```suggestion
As you can see, the AnonCreds module takes a list of registry modules. These modules will be used to resolve the AnonCreds objects. We recommend [Indy VDR](./indy-vdr) as an AnonCreds registry for Hyperledger Indy networks, but [Indy SDK](./indy-sdk) can also be used.
```

# Anoncreds-rs

> [AnonCreds](https://github.com/hyperledger/anoncreds) is a direct

Reviewer: This link 
```suggestion
[AnonCreds](https://github.com/hyperledger/anoncreds-rs) is a direct
```

implementation of the [AnonCreds V1.0
specification](https://hyperledger.github.io/anoncreds-spec/) that provides
functionality like; creating a schema object, creating a credential definition

> object, creating a credential, verifying a proof presentation and much more.

Reviewer: Although it works in the rendered html, adding a line breaks, breaks content in the preview of other pages when linked.

(see image in PR review message)

:::caution

> Support for AnonCreds in Aries Framework JavaScript is currently experimental.

Reviewer: ```suggestion
Support for the AnonCreds RS library in Aries Framework JavaScript is currently experimental.
```

We recommend new projects to use AnonCreds from the start, and also to migrate

> existing projects to Anoncreds, from the indy-sdk. However, projects may

Reviewer: You get it ;)

```suggestion
existing projects to AnonCreds RS, from the Indy SDK. However, projects may
```

experience some issues. If you encounter any issues, please [open an
issue](https://github.com/hyperledger/aries-framework-javascript/issues/new).

Currently, there are few limitations to using AnonCreds.

- React Native projects leveraging [Expo](https://expo.dev) are not supported
  at the moment. Support for Expo will be added soon.
- When running in Node.JS, only Node.JS 18 is suported for now. See [Supported
  Node.JS versions for AnonCreds](#supported-nodejs-versions-for-anoncreds)

:::

> ### Installing AnonCreds

Reviewer: This guide is for setting up AnonCreds RS right? How does it relate to setting up AnonCreds in general? Is that handled by the specific tutorials, or do we also need to update the indy sdk tutorial to add the anoncreds module?

Reviewer 2: Yes, I would say just anoncreds-rs. I think if we want to add some context surrouding the anoncreds package, we should do it as a `concept` and this would just be a minimal setup for an agent using the anoncredsrs library.

> When using Aries Framework JavaScript with AnonCreds, there are a few extra

Reviewer: We need to be careful in using AnonCreds and AnonCreds RS.

> dependencies that need to be installed. We need to install the
> `@hyperledger/anoncreds`, which contains the interfaces, and

Reviewer: ```suggestion
dependencies that need to be installed. We need to install
`@hyperledger/anoncreds`, which contains the interfaces, and
```

OR 

```suggestion
dependencies that need to be installed. We need to install the
`@hyperledger/anoncreds` package, which contains the interfaces, and
```

`@hyperledger/anoncreds-rs` which is an implementation which depends on a
wrapper of anoncreds-rs. Secondly, we need to add native bindings for the

> specific platform `@hyperledger/anoncreds-<platform>`. Currently there are

Reviewer: Can you keep the installation as consistent as possible? So if modifying this by adding the `nodejs` etc.., we should add it to the other existing setup guides as well.

> specific platform `@hyperledger/anoncreds-<platform>`. Currently there are
> bindings for Node.JS, `nodejs` and React Native, `react-native`.

Reviewer: ```suggestion
specific platform `@hyperledger/anoncreds-<platform>`. Currently there are
bindings for Node.JS (`nodejs`) and React Native (`react-native`).
```

Or maybe:

```suggestion
specific platform `@hyperledger/anoncreds-<platform>`. Currently there are
bindings for Node.JS as `@hyperledger/anoncreds-nodejs`, and React Native as `@hyperledger/anoncreds-react-native`.
```

<!--tabs-->

# Node.JS

```console
yarn add @aries-framework/anoncreds @aries-framework/anoncreds-rs @hyperledger/anoncreds-nodejs
```

# React Native

```console
yarn add @aries-framework/anoncreds @aries-framework/anoncreds-rs @hyperledger/anoncreds-react-native
```

<!--/tabs-->

### Adding AnonCreds to the agent

After installing the dependencies, we can register both the AnonCreds and
AnonCredsRs module on the agent.

<!--tabs-->

# Node.JS

```typescript showLineNumbers set-up-anoncreds.ts section-1

```

# React Native

```typescript showLineNumbers set-up-anoncreds-rn.ts section-1

```

<!--/tabs-->

As you can see, the AnonCreds module takes a list of registry modules. These
modules will be used to resolve the AnonCreds objects.

### Supported Node.JS versions for AnonCreds

Due to an issue in `ref-napi` (which is used in the Node.JS bindings for

> AnonCredsRs), performance for AnonCredsRs in Node.JS is not as expected. A

Reviewer: ```suggestion
AnonCreds RS), performance for AnonCreds RS in Node.JS is not as expected. A
```
